docs/DeveloperGuide.md

```diff
@@ -9,32 +9,129 @@
+- **Scalability**: By using `FinancialEntry` as a base class, `FinancialList` can easily manage new types of entries (e.g., `Investment` or `Loan`), ensuring the design is open to future extensions.
 
-### Implementation Details
+---
+#### FinancialEntry Component
+__Overview__
+
+`FinancialEntry` is an abstract base class that represents a generic financial record.
+It defines shared attributes such as `amount`, `description`, and `date`, which are common across both `Income` and `Expense`.
+`Income` and `Expense` inherit these properties and methods, each adding specific functionality related to its type.
+
+__Implementation__
+
+The class diagram above shows `FinancialEntry` as the base class with `Income` and `Expense` as specific implementations.
+{input diagram here}
+
+__Class Structure__
+- **Attributes**:
+  - `amount`: `double` — Represents the monetary value of the entry.
+  - `date`: `LocalDate` — The date associated with the transaction.
+  - `description`: `String` — A description identifying the entry.
+
+__Constructor__
+
+The `FinancialEntry` constructor initializes `amount`, `description`, and `date`.
+
+- **Key Arguments**:
+  - `double amount`: Monetary value for the entry.
+  - `String description`: Description or label for the entry.
+  - `LocalDate date`: Date of the entry.
+
+__Methods__
+
+- **Core Methods** (inherited by both `Income` and `Expense`):
+  - `getAmount()`, `getDescription()`, `getDate()`: Accessor methods for each attribute.
+  - `setAmount(double newAmount)`, `setDescription(String newDescription)`, `setDate(LocalDate newDate)`: Mutator methods for updating values.
+
+- **Custom Methods for Income and Expense**:
+  - **toString()**:
+    - `Income`: Returns formatted string as `[Income] - description $amount (on date)`.
+    - `Expense`: Returns formatted string as `[Expense] - description $amount (on date)`.
+  - **toStorageString()**:
+    - `Income`: Formats as `"I | amount | description | date"` for storage.
+    - `Expense`: Formats as `"E | amount | description | date"` for storage.
+
+__Usage Example__
+
+The following code segment demonstrates the creation of `Income` and `Expense` entries:
+```
+Income income = new Income(500.00, "Freelance Project", LocalDate.of(2023, 10, 27));
+Expense expense = new Expense(50.00, "Groceries", LocalDate.of(2023, 10, 28));
+
+System.out.println(income.toString());
+System.out.println(expense.toString());
+```
+
+__Design Considerations__
+- **Future Extension**: Additional fields could be added to `Income` and `Expense` for more specific details, such as a `source` for `Income` or a `category` for `Expense`.
+- **Abstract Base Class**: The design decision to make `FinancialEntry` abstract enables extensibility, allowing for new types of financial records without modifying `FinancialList` or existing subclasses.
```

Comment:
This statement doesn't seem entirely correct, the base class doesn't need to be abstract to allow for extension right?